### ISSUE TITLE
Masking support for multiple values.

### ISSUE THREAD

**Issue by vyshnavigutta369** [tensorflow/tensorflow#43708]
Is there a way to mask more than one value in input before sending to Keras layers in TensorFlow? I have to pad in-between too with 0 to make the data balanced in addition to post padding. But my Seq2Seq model is unable to distinguish the in-between paddings and the paddings at the end and during testing, the output from decoder stops as soon as the first in-between padding occurs.

Consider an input tensor to be,

    [[1,2,3,4], [5,6,7]] 

Below is the way I am padding the above tensor. Each sub-list must be of size 6 and the total number of sub-lists should be 3.

    [ 3,4,5,6,0,0  , 7,8,9,0,0,0  , 0,0,0,0,0,0 ]

I  want my tensor to be masked this way, where 2 is also masked along with 0, before sending to subsequent layers like embedding and RNN layers.

    [ 3,4,5,6,2,2  , 7,8,9,2,2,2,  0,0,0,0,0,0 ]

Could some one also point out a possible solution for preventing the decoder output from stopping at the end of processing the first sub-list of an input?

**Comment by klausk55** [comment]
@vyshnavigutta369
I am not sure whether I have understood you right or not, but maybe two consecutive masking layers (one with mask_value=0. and the other with mask_value=2.) is what you are looking for.

**Comment by SuryanarayanaY** [comment]
Hi @vyshnavigutta369 ,

Sorry that Iam unable to catch your requirement here.
Could you please go through the attached documentation links [masking & padding](https://www.tensorflow.org/guide/keras/masking_and_padding) and also refer these links [Masking](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Masking) and [Embedding](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Embedding) and let us know if it helps.
If you are still not getting your requirement fulfilled please come back again with a better use case to make us understand your requirement clearly.

Thankyou!
